content/blaze.md

```diff
@@ -0,0 +1,559 @@
+
+Note that simple string interpolations like this will always escape any HTML for you---so you don't need to perform safety checks for XSS.
+
+Additionally we can see an example of a *template helper*---`{{checkedClass}}` calls out to the `checkedClass` helper defined in a separate JavaScript file, which combined with the template defines the `todosItem` component:
+
+```js
+Template.todosItem.helpers({
+  checkedClass() {
+    return this.todo.checked && 'checked';
+  }
+});
+```
+
+In the context of a Blaze helper, `this` is scoped to the current current *data context* at the point the helper was used. This can be hard to reason about, so it's often a good idea to instead pass the required data into the helper as an argument.
+
+Apart from simple interpolation, mustache tags can control flow of the template. For instance, in the `listsShow` template, we render a list of todos via:
+
+```html
+  {{#each todo in todos}}
+    {{> todosItem (todoArgs todo)}}
+  {{else}}
+    <div class="wrapper-message">
+      <div class="title-message">No tasks here</div>
+      <div class="subtitle-message">Add new tasks using the field above</div>
+    </div>
+  {{/each}}
+```
+
+This snippet illustrates a few things:
+
+ - The `{{#each in}}` block helper which renders a block one per element in an array or cursor, or an `{{else}}` block if no items exist.
+ - Template inclusion `{{> todosItem (todoArgs todo)}}` which renders the `todosItem` component with a set data context, based on the output of the `todosArg` helper.
+
+You can read about the full Spacebars syntax [here](https://github.com/meteor/meteor/blob/devel/packages/spacebars/README.md), but in this section we'll attempt to cover some important finer details.
+
+<h3 id="data-contexts">Data contexts and lookup</h3>
+We've seen `{{todo.title}}` accesses the `title` property of the `todo` item on the current data context. Additionaly, `..` access the parent data context (rarely a good idea), `list.todos.[0]` accesses the first element of the `todos` array on `list`.
+
+Also, note that Spacebars is very forgiving of `null` values. It will not complain if you try to access a property on a `null` value (for instance `foo.bar` if `foo` is not defined), but instead simply treat it also as null. However there are exceptions to this---trying to call a `null` function, or doing the same *within* a helper will lead to exceptions.
+
+<h3 id="helpers">Calling helpers with arguments</h3>
+You can provide arguments to a helper like `checkedClass` by simply placing the argument after the helper call, as in: `{{checkedClass todo true 'checked'}}`. You can also provide a list of named keyword arguments to a helper with `{{checkedClass todo noClass=true classname='checked'}}. In this case, you might access those arguments with:
```

Comment:
I'd specifically call out `kws.hash` in the text. It's a bit odd and deserves clarification. I'd even go so far as to suggest avoiding keyword arguments to helpers for clarity?